### ISSUE TITLE
Query with multiple query_string generates invalid query

### ISSUE THREAD

**Issue by damieng** [ElasticLINQ/ElasticLINQ#43]
Similar to bug #42 but where AND or OR just references other contains statements we should generate a combined query_string using the enhanced rules/syntax at http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html

``` C#
context.Query<Product>().Query(g => (g.name.Contains("a") && 
g.name.StartsWith("b")) || g.something == "theWord");
```

Would then become:

``` json
{
    "query_string": {
      "query": "(name:*a* AND name:b*) OR (something:theWord)"
    }
}
```

**Comment by damieng** [comment]
This will require some significant expansion on the query_string handling including proper escaping and wildcard mapping. 

A shorter-term solution for 1.2 might be to just support AND for multiple terms against the same field and add a helpful error message for other scenarios.

**Comment by doubret** [comment]
can't you simply generate several filters and combine them with regular AND and OR filters ?

**Comment by damieng** [comment]
Rather than support the embedded syntax we are just combining them into a bool query. Fixed with #34.
